src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,82 @@
+package duke;
+
+/**
+ * Parser class parses the commands given by the users.
+ * @author Dominic Siew Zhen Yu
+ */
+
+public class Parser {
+
+    private String taskName;
+    private String command;
+    private String deadlineOrTimeline;
+
+    /**
+     * constructor for the Parser class
+     */
+
+    public Parser() {
+    }
+
+    /**
+     * parsing(String input) takes in a string from the given input and parses the string
+     * into the taskName, command and deadlineOrTimeline variables.
+     *
+     * @param input the given input by the user
+     */
+
+    public void parsing(String input) {
```

Comment:
Perhaps a better name for this method will be `parse`? I think we generally should avoid adding '-ing' behind our methods and class names.